tests/test_core.py

```diff
@@ -1008,18 +1005,17 @@ def test_class_properties_lifecycle_strong(self):
         class StrongProperties(NSObject):
             object = objc_property(ObjCInstance)
 
-        pool = NSAutoreleasePool.alloc().init()
+        with autoreleasepool():
 
-        properties = StrongProperties.alloc().init()
+            properties = StrongProperties.alloc().init()
 
-        obj = NSObject.alloc().init()
-        obj_pointer = obj.ptr.value  # store the object pointer for future use
+            obj = NSObject.alloc().init()
+            obj_pointer = obj.ptr.value  # store the object pointer for future use
 
-        properties.object = obj
+            properties.object = obj
 
-        del obj
-        del pool
-        gc.collect()
+            del obj
+            gc.collect()
```

Comment:
If you want to match the behavior of the previous code, the `gc.collect()` call needs to be outside of the `with` block, so that it runs *after* the autorelease pool has been popped:

```suggestion

        gc.collect()
```

Reply:
Yes, this change was deliberate. The idea is to delete the Python object first, then trigger a garbage collection to make sure that its `__del__` method is called, then clear the autoreleasepool to clean up any autoreleased objects.

The order does not really matter unless one of the Objective-C objects is kept alive by a Python reference *and* was autoreleased previously. The fact that the tests passed before this change means that the garbage collection itself probably is not required at all.

Reply:
Thinking about it again, the logic behind the previous code was probably that `gc.collect()` comes after all of the `del` calls, to try to force Python to deallocate the `del`ed objects right away - especially the autorelease pool, which only got released once its `ObjCInstance` was deallocated on the Python side. Now that the autorelease pool is explicitly released by the context manager, this is no longer a problem, so the `gc.collect()` call is probably safe to go inside the context manager as you say. (I wouldn't remove the `gc.collect()` completely, because there's always the possibility that a simple `del` might not deallocate the object right away for some reason.)

Reply:
Right, that makes sense. The order did matter previously, because for the autorelease pool to work its Python reference needed to be deallocated, but this is no longer the case.

Reply:
Right, that makes sense. The order did matter previously, because for the autorelease pool to work its Python reference needed to be deallocated, but this is no longer the case.